snapcraft/internal/states/_global_state.py

```diff
@@ -15,12 +15,24 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import os
-from typing import Dict, List, Type
+from typing import List, Optional, Type
+
+from mypy_extensions import TypedDict
 
 from snapcraft import yaml_utils
 from snapcraft.internal.states._state import State
 
 
+StateDict = TypedDict(
```

Comment:
Not asking to change this here, but I think at this rate, it may just make sense to promote "build-packages", "build-snaps", and "required-grade" as typed properties of GlobalState.  Seems like unnecessary complexity.  The use of TypedDict strongly suggests a python object that can be marshaled/unmarshaled.  I thought you didn't like using the typed extensions anyhow?? :)

Reply:
I never said I did not like the typed extension, did I?

Remember that GlobalState is marshalled to a yaml object (!!GlobalState).